directories/src/lib.rs

```diff
@@ -15,6 +15,17 @@
 #![deny(missing_docs)]
 #![warn(rust_2018_idioms)]
 
+// FIXME(MSRV >= 1.40.0): rust 1.40.0 stabilized `Option::as_deref`. Starting with rust 1.42.0,
```

Comment:
I wasn't sure how to format the comment to indicate that this comment and the associated `#![allow()]` should be removed when MSRV is greater than or equal to 1.40.0.
I think this way makes sense and is easy enough to understand.